tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/UsageReport/WriteUsageBurndownData.cs

```diff
@@ -0,0 +1,124 @@
+        /// Specifies the path to the prebuilt baseline file
+        /// to be used to generate the burndown.
+        /// </summary>
+        [Required]
+        public string PrebuiltBaselineFile { get; set; }
+
+        /// <summary>
+        /// Output data CSV file. 
+        /// </summary>
+        [Required]
+        public string OutputFilePath { get; set; }
+
+        class commit
+        {
+            public string sha { get; set; }
+            public string title { get; set; }
+            public DateTime commitDate { get; set; }
+            public int packageVersionCount { get; set; }
+            public int packageCount { get; set; }
+
+            public override string ToString()
+            {
+                return $"{sha}, {title}, {commitDate}, {packageVersionCount}, {packageCount}";
+            }
+        }
+
+        public override bool Execute()
+        {
+            string baselineRelativeFileName = PrebuiltBaselineFile.Replace(RootDirectory, "");
+            string gitLogCommand = $"log --first-parent --pretty=format:%H,%f,%ci -- {PrebuiltBaselineFile}";
+
+            DateTime startTime = DateTime.Now;
+            Log.LogMessage(MessageImportance.High, "Generating summary usage burndown data...");
+
+
+            var data = ExecuteGitCommand(RootDirectory, gitLogCommand).AsParallel().Select(commitLine =>
+            {
+                var splitLine = commitLine.Split(',');
+                var commit = new commit() { sha = splitLine[0], title = splitLine[1], commitDate = DateTime.Parse(splitLine[2]) };
+                var fileContents = GetFileContents(baselineRelativeFileName, commit.sha);
+                var usages = UsageData.Parse(XElement.Parse(fileContents)).Usages.NullAsEmpty().ToArray();
+                commit.packageVersionCount = usages.Count();
+                commit.packageCount = usages.GroupBy(i => i.PackageIdentity.Id).Select(grp => grp.First()).Count();
+                return commit;
+            })
+            .Select(c => c.ToString())
+            .Aggregate((a, b) => a + '\n' + b);
+    
+            Directory.CreateDirectory(Path.GetDirectoryName(OutputFilePath));
+
+            File.WriteAllText(OutputFilePath, data);
+
+            Log.LogMessage(
+                MessageImportance.High,
+                $"Generating summary usage burndown data at {OutputFilePath} done. Took {DateTime.Now - startTime}");
+
+            return !Log.HasLoggedErrors;
+        }
+
+        /// <summary>
+        /// Get the contents of a git file based on the commit sha.
+        /// </summary>
+        /// <param name="relativeFilePath">The relative path (from the git root) to the file.</param>
+        /// <param name="commitSha">The commit sha for the version of the file to get.</param>
+        /// <returns>The contents of the specified file.</returns>
+        public string GetFileContents(string relativeFilePath, string commitSha)
+        {
+            WebClient client = new WebClient();
```

Comment:
Is `WebClient` like `HttpClient` in that you should just use one instance where you can? I think `new`ing for each request creates a scaling concern because each one takes control of a port, something like that.